lib/axon/compiler.ex

```diff
@@ -170,6 +172,8 @@ defmodule Axon.Compiler do
           expr
       end
     end
+
+    &Nx.Defn.jit_or_apply(predict_fn, [&1, &2])
```

Comment:
@josevalim So previously I was not jitting anything when these functions we're called, but I think this is probably the better way to go as I think this will still work in a nested JIT (e.g. if you wanted to override the default compiler options) as well as within defn.

Reply:
Agreed! :+1: